### PR TITLE
add swap pallet

### PR DESCRIPTION
**Pull request by nuel77**
- adds swap pallet for polkapool for https://github.com/Polkadex-Substrate/thea/issues/50

### CODE REVIEW

File: pallets/router/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+[package]
+authors = ["Polakdex Authors"]
```
Comment (serhii-temchenko) [inline pallets/router/Cargo.toml]: Typo

File: pallets/support/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+[package]
+authors = ["Polakdex Authors"]
```
Comment (serhii-temchenko) [inline pallets/support/Cargo.toml]: Typo

File: pallets/router/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+
+// mod benchmarking;
+//
+// #[cfg(test)]
+// mod mock;
+// #[cfg(test)]
+// mod tests;
+
+// pub mod weights;
+// pub use weights::WeightInfo;
```
Comment (serhii-temchenko) [inline pallets/router/src/lib.rs]: Please remove

File: pallets/swap/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,1166 @@
...
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// #[cfg(test)]
+// mod mock;
+// #[cfg(test)]
+// mod tests;
+//
+// mod benchmarking;
+// pub mod weights;
```
Comment (serhii-temchenko) [inline pallets/swap/src/lib.rs]: Same

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/router/src/lib.rs (hunk 3 above)
Comment (zktony) [reply pallets/router/src/lib.rs]: done

**Review by serhii-temchenko — DISMISSED** [review]

**Review by serhii-temchenko — DISMISSED** [review]

**Pull request merged**